ASoC: Intel: soc-acpi-intel-ptl-match: add rt712_vb_l3_rt1320_l2 and rt713_vb_l3_rt1320_l12 support

ujfalusi opened the pull request:
Add rt712_vb on SDW link 3 and 1 rt1320 on SDW link 2 configuration support.

ujfalusi committed:
ASoC: Intel: soc-acpi-intel-ptl-match: add rt712_vb_l3_rt1320_l2 support

Add rt712_vb on SDW link 3 and 1 rt1320 on SDW link 2 configuration
support.

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@linux.intel.com>

bardliao commented on sound/soc/intel/common/soc-acpi-intel-ptl-match.c:
```diff
@@ -206,6 +215,15 @@ static const struct snd_soc_acpi_adr_device rt1320_1_group2_adr[] = {
+	{
+		.adr = 0x000230025d132000ull,
+		.num_endpoints = 1,
+		.endpoints = &single_endpoint,
```
We should use group 1 as rt1320 works with rt712_vb's amp. IOW, rt712_vb and rt1320 are aggregated.

ujfalusi commented:
Changes since v1:
- use group1 for rt1320

ujfalusi committed:
ASoC: Intel: soc-acpi-intel-ptl-match: add rt713_vb_l3_rt1320_l12 support

Add rt713_vb on SDW link 3 and 2 rt1320 on SDW link 1 and 2 configuration
support.

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@linux.intel.com>

ujfalusi commented:
Changes since v2:
- Addresses should use capital letters in hex numbers
- corrected rt1320 address to production version
- added new config for rt713-l3 with rt3120-l12

ujfalusi commented on sound/soc/intel/common/soc-acpi-intel-ptl-match.c:
```diff
@@ -388,6 +425,13 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_ptl_sdw_machines[] = {
 	},
+	{
+		.link_mask = BIT(1) | BIT(2) | BIT(3),
+		.links = ptl_sdw_rt713_vb_l3_rt1320_l12,
```
Hrm, either the commit message (rt712) or the patch is not correct (rt713), need to check.

ujfalusi commented:
Changes since v3:
- Fix commit message for rt713_l3-rt1320_l12

plbossart commented on sound/soc/intel/common/soc-acpi-intel-ptl-match.c:
```diff
@@ -206,6 +215,15 @@ static const struct snd_soc_acpi_adr_device rt1320_1_group2_adr[] = {
+		.adr = 0x000230025D132001ull,
+		.num_endpoints = 1,
+		.endpoints = &spk_r_endpoint,
+		.name_prefix = "rt1320-1"
```
not sure I follow why this one needs to have the '-1' suffix for group 1 but the rt712 above does not. IIRC the -1 and -2 where related to amps, not groups.

bardliao replied:
The '-1' suffix will be used by the ucm. https://github.com/alsa-project/alsa-ucm-conf/blob/master/ucm2/sof-soundwire/rt1320.conf. It assume the '-1', '-2' prefix in the control names.

bardliao commented:
Not sure what happened to the LNL SDW test https://sof-ci.01.org/linuxpr/PR5315/build6111/devicetest/index.html. Let rerun the test.

bardliao commented:
SOFCI TEST

lgirdwood commented:
> Not sure what happened to the LNL SDW test https://sof-ci.01.org/linuxpr/PR5315/build6111/devicetest/index.html. Let rerun the test.

This seems unrelated to PTL, but could we be including a shared topology file that breaks ?
```
2025-02-14 02:44:01 UTC [ERROR] Caught kernel log error
===========================>>
[    5.740282] kernel: sof-audio-pci-intel-lnl 0000:00:1f.3: error: can't connect DAI alh-copier.SDW2-Playback.0 stream SDW2-Playback
[    5.740285] kernel: sof-audio-pci-intel-lnl 0000:00:1f.3: failed to add widget type 27 name : alh-copier.SDW2-Playback.0 stream SDW2-Playback
[    5.740294] kernel: sof_sdw sof_sdw: ASoC: failed to load widget alh-copier.SDW2-Playback.0
[    5.740297] kernel: sof_sdw sof_sdw: ASoC: topology: could not load header: -22
[    5.740353] kernel: sof-audio-pci-intel-lnl 0000:00:1f.3: error: tplg component load failed -22
[    5.740362] kernel: sof-audio-pci-intel-lnl 0000:00:1f.3: error: failed to load DSP topology -22
[    5.740365] kernel: sof-audio-pci-intel-lnl 0000:00:1f.3: ASoC: error at snd_soc_component_probe on 0000:00:1f.3: -22
[    5.740407] kernel: sof_sdw sof_sdw: ASoC: failed to instantiate card -22
[    5.740863] kernel: sof_sdw sof_sdw: error -EINVAL: snd_soc_register_card failed -22
[    5.740869] kernel: sof_sdw sof_sdw: probe with driver sof_sdw failed with error -22
```

ujfalusi commented:
> > Not sure what happened to the LNL SDW test https://sof-ci.01.org/linuxpr/PR5315/build6111/devicetest/index.html. Let rerun the test.
> 
> This seems unrelated to PTL, but could we be including a shared topology file that breaks ?
> 
> ```
> 2025-02-14 02:44:01 UTC [ERROR] Caught kernel log error
> ===========================>>
> [    5.740282] kernel: sof-audio-pci-intel-lnl 0000:00:1f.3: error: can't connect DAI alh-copier.SDW2-Playback.0 stream SDW2-Playback
> [    5.740285] kernel: sof-audio-pci-intel-lnl 0000:00:1f.3: failed to add widget type 27 name : alh-copier.SDW2-Playback.0 stream SDW2-Playback
> [    5.740294] kernel: sof_sdw sof_sdw: ASoC: failed to load widget alh-copier.SDW2-Playback.0
> [    5.740297] kernel: sof_sdw sof_sdw: ASoC: topology: could not load header: -22
> [    5.740353] kernel: sof-audio-pci-intel-lnl 0000:00:1f.3: error: tplg component load failed -22
> [    5.740362] kernel: sof-audio-pci-intel-lnl 0000:00:1f.3: error: failed to load DSP topology -22
> [    5.740365] kernel: sof-audio-pci-intel-lnl 0000:00:1f.3: ASoC: error at snd_soc_component_probe on 0000:00:1f.3: -22
> [    5.740407] kernel: sof_sdw sof_sdw: ASoC: failed to instantiate card -22
> [    5.740863] kernel: sof_sdw sof_sdw: error -EINVAL: snd_soc_register_card failed -22
> [    5.740869] kernel: sof_sdw sof_sdw: probe with driver sof_sdw failed with error -22
> ```

Someone left the device in mockup mode?
```
Feb 14 02:42:03 jf-lnlm-rvp-sdw-1 kernel: snd_soc_sdw_utils:asoc_sdw_count_sdw_endpoints: sof_sdw sof_sdw: Found 3 devices with 3 endpoints
Feb 14 02:42:03 jf-lnlm-rvp-sdw-1 kernel: snd_soc_sdw_utils:asoc_sdw_parse_sdw_endpoints: sof_sdw sof_sdw: Adding prefix sdw_mockup_headset0 for sdw:0:0:0105:aa55:00
Feb 14 02:42:03 jf-lnlm-rvp-sdw-1 kernel: snd_soc_sdw_utils:asoc_sdw_parse_sdw_endpoints: sof_sdw sof_sdw: Add dev: 0, 0x105aa5500 end: 0, dai: 0, P/C to solo: 0
Feb 14 02:42:03 jf-lnlm-rvp-sdw-1 kernel: snd_soc_sdw_utils:asoc_sdw_parse_sdw_endpoints: sof_sdw sof_sdw: Adding prefix sdw_mockup_amp1 for sdw:0:1:0105:55aa:00
Feb 14 02:42:03 jf-lnlm-rvp-sdw-1 kernel: snd_soc_sdw_utils:asoc_sdw_parse_sdw_endpoints: sof_sdw sof_sdw: Add dev: 1, 0x100010555aa00 end: 0, dai: 1, P/C to solo: 0
Feb 14 02:42:03 jf-lnlm-rvp-sdw-1 kernel: snd_soc_sdw_utils:asoc_sdw_parse_sdw_endpoints: sof_sdw sof_sdw: Adding prefix sdw_mockup_mic3 for sdw:0:3:0105:5555:00
Feb 14 02:42:03 jf-lnlm-rvp-sdw-1 kernel: snd_soc_sdw_utils:asoc_sdw_parse_sdw_endpoints: sof_sdw sof_sdw: Add dev: 3, 0x3000105555500 end: 0, dai: 2, -/C to solo: 0
Feb 14 02:42:03 jf-lnlm-rvp-sdw-1 kernel: snd_soc_sof_sdw:sof_card_dai_links_create: sof_sdw sof_sdw: DAI link numbers: sdw 5, ssp 0, dmic 0, hdmi 3, bt: 0
```

@kv2019i ?

bardliao commented:
@ujfalusi That's my fault. I used `jf-lnlm-rvp-sdw-1` to test SDW mockup mode yesterday. But I removed the mockup ADR from the acpi table already. Not sure why mockup devices are still present.

bardliao commented:
The aml is not in the `/lib/firmware/acpi-upgrades/` folder already.
```
ubuntu@jf-lnlm-rvp-sdw-1:~$ ls /lib/firmware/acpi-upgrades/
ubuntu@jf-lnlm-rvp-sdw-1:~$
```

I did it by the `sudo ./acpi-rm` command. Did I miss something?

ujfalusi commented:
I have manually removed them, rebuilt the initramfs and rebooted the machine

ranj063 commented:
SOFCI TEST
